src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -1327,6 +1327,27 @@ public void Add(IEnumerable<T> items)
         }
 
         /// <summary>
+        /// Special add for TypeTable type entries that removes redundant file entries.
+        /// </summary>
+        internal void AddTypeTableTypesInfo(IEnumerable<T> items)
+        {
+            if (!(items is IEnumerable<SessionStateTypeEntry>)) { throw new PSInvalidOperationException(); }
+
+            lock (_syncObject)
+            {
+                foreach (var element in items)
+                {
+                    var typeEntry = element as SessionStateTypeEntry;
+                    if (typeEntry.TypeData != null)
+                    {
+                        // Skip type file entries.
```

Comment:
Ok, I understand you now.  I agree that this is a non-issue, not necessarily because changing a shared type table is not allowed but because the TypeTable is consistent in how it stores types.  But I'll add a test for this.